cmd/kubelet/app/server.go

```diff
@@ -425,7 +425,9 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 			if err != nil {
 				return err
 			}
-			if cloud == nil {
+			if strings.EqualFold(s.CloudProvider, cloudprovider.ExternalCloudProvider) {
```

Comment:
I've added the `if` here because I dont want to confuse the user by saying "No cloudprovider specified" when they have specified "external" explicitly.

It's ok to call InitCloudProvider() above, because there is logic in pkg/cloudprovider/plugins.go that handles "external" cloudprovider from a central location. Handling the "external" cloudprovider logic there will make will make the UX consistent for  other binaries - controller-manager and api-server as well, where the user can specify "external" as the cloudprovider and expect the cloud-controller-manager to take care of every cloudprovider call.